confignode/src/main/java/org/apache/iotdb/confignode/persistence/Snapshot.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.confignode.persistence;
+
+import org.apache.thrift.TException;
+
+import java.io.File;
+import java.io.IOException;
+
+public interface Snapshot {
```

Comment:
Maybe we could put this interface in the IStateMachine interface of the consensus module, and make a snapshotAPI like the EventAPI, which would be easier to maintain.

Reply:
We can consider doing so, but this snapshot is  where it will actually be processed _(It is not the interface for the consensus group to interact with the state machine)_ .  This seems beyond the scope of `IstateMachine`.

Reply:
If you think they have a different function you can change the name or add a comment, otherwise it will be ambiguous

Reply:
If you think they have a different function you can change the name or add a comment, otherwise it will be ambiguous

Reply:
OK，let me update the name and add some comments

Reply:
already updated